Improve Group Create and Edit

ardelato opened the pull request:
## Description

This modifies the Groups functionality to improve and fix the creation and edit page. The core issues stemmed from misunderstandings with the UI.

There were two major changes which revolved around the website field and the Postcode/Timezone fields.

#### Prefix Group Website
The website validation expects for the url to include `https://` so if folks pass www.website.com, it would only state the url was invalid but not not elaborate why. This expectation is now surfaced and automated by adding the protocol as part of the website field.

|Before|After|
|:---:|:---:|
|  ![image](https://github.com/user-attachments/assets/91000dce-c524-48cf-92ae-107db76baddf)| ![image](https://github.com/user-attachments/assets/22d8e8cf-3765-4108-ab34-a17a05cb7d75)|

We will still save the websites with the protocol but we remove the responsibility for the user to declare it and explicitly communicate it will be set.

#### Override Checkboxes

This is a multi-layered change that addresses a misunderstanding of the `Group Postcode` field and `Group Timezone` field.

These fields essentially override their respective values determined by the `Group Location` and the `Network Session`. However, it was not clear it is what they did. 

The `Group Postcode` field was disabled when creating a group and only enabled when editing a group as an admin. As such, it first seemed like a broken text field as all other fields did not have this condition.

The `Group Timezone` field was enabled from the start, but small text under the field was the only thing that indicated an override change.

Given their core behavior I introduced override checkboxes that would `enable` these fields to be overridden. To further deepen this intention, I added event handlers for both fields to get updated when the `Group Location` changed. Indicating that the fields are automatically set by the value of `Group Location`.

Doing this also required a DB migration in order to save the group settings.


|No Override|Override|
|:---:|:---:|
|   ![SnagitHelper2023 2025-05-12 17 07 47](https://github.com/user-attachments/assets/891b96b9-dd04-430e-b6fe-05829fa9a9fb)    | ![image](https://github.com/user-attachments/assets/f294d210-0ba0-4b65-ae49-7e455a56486d)      |

### CR Notes

We will probably want approval from Amber that this resolved some original issues she found.

qa_req 0 Its deployed on https://restarters.eks.ubreakit.com

ardelato committed:
feat(groups): update logic for group postcode field

Let's allow anyone to edit the postcode field. This means it won't
be disabled when creating a new group and when editing an existing
group while not being an admin.

I thought about just not rendering the postcode field at all but I did
not see the same restrictions to the other fields. As such, it made
more sense to me to allow anyone to edit the postcode field like they
could the other fields.

As part of this change, I updated the hint text to make it clear that
the postcode field will take precedence over the location's postcode. In
addition, when first creating a group, the postcode field will be
pre-filled with the location's postcode.

ardelato committed:
feat(timezone): add TimeZoneController and integrate timezone lookup

Introduced a new TimeZoneController to handle timezone lookups based on
latitude and longitude. This will ensure we don't expose the Google Maps
API key to the frontend.

By utilizing this new route, we can set the timezone for a group based
on its location, rather then the app's network.

ardelato committed:
feat(groups): add override options for postcode and timezone

Enhanced the Group model and related components to support 'override_postcode'
and 'override_timezone' features. This will give us more control over the
group creation page as the current logic is somewhat confusing.

For example, if the fields are empty then the location's postcode and timezone
will be used. Afterwhich, if the location is changed, the postcode and timezone
will not be updated.

While we could update the fields every time the location is changed, this
would regress the ability to override the postcode and timezone.

As such, let's implement the "override" functionality in a more explicit
way by adding a checkbox to state whether the postcode and timezone should
be overridden.

ardelato committed:
feat(groups): enhance readonly behavior for postcode and timezone inputs

Updated the GroupLocation and GroupTimeZone components to conditionally set
the readonly attribute on the postcode and timezone input fields based on
the respective override checkboxes. This ensures that users can only edit
the fields when the override option is enabled, improving the user experience
and clarity of the form.

This won't affect the existing functionality of the automatic setting
of the postcode and timezone fields when the location is changed.

ardelato committed:
fix(groups): enable two-way binding for override postcode in GroupAddEdit

Updated the GroupAddEdit component to use two-way binding for the overridePostcode
prop. Even if we checked the override box for the Postcode field, the value
would still be modified when the location was changed and the checkbox
would be unchecked.

ardelato committed:
fix(groups): adjust grid row for responsive layout in GroupAddEdit component

The grid row for the map was not being set correctly for the breakpoint.
It was overlapping with the previous element's row that was set to 1/4.

ardelato committed:
fix(groups): ensure boolean conversion for override postcode and timezone inputs

The requests had boolean values as strings, so we need to convert them to
actual booleans to be able to save them in the database.

ardelato committed:
fix(groups): allow website input without https:// prefix

The website input field was not allowing the user to input a website
without the https:// prefix. This change allows the user to input a
website without the https:// prefix and also communicates to the user
that the website will be prefixed with https://.

mlahargou reviewed (approved):
CR 📱 But maybe it makes sense to handle this on the PHP side. It's pretty easy to add `https://` if it doesn't exist.

(I am assuming that `this.$emit('update:website', url)` calls an api endpoint on the php backend.)

ardelato commented:
deploy_block ⚡ on Amber's approval

ardelato commented:
> (I am assuming that this.$emit('update:website', url) calls an api endpoint on the php backend.)

The emit function signals for the parent component to update its data prop so it allows two way sync of the value between the parent GroupAddEdit and GroupWebsite. This is all frontend.

While this all could be done in the backend I left it in the frontend since I wasn't sure how much this would affect the backend side. This quarantines the changes to just the frontend since we will still be sending the same expected value to the backend.

ardelato committed:
refactor(groups): use request boolean func instead of filter_var

This slipped through the cracks and was supposed to be part of
a13202d46e554816ac5e76efb14ac3c6ba9b708c. I had first used filter_var
but then learned that we could use the request boolean function instead
as part of Laravel 10+. I had updated one instance but not the other.

mlahargou reviewed (approved):
CR 📱

ardelato commented:
un_deploy_block ⚡ I am going to merge this in. If there are any changes that need to be made we can do so after the fact.
